src/Paginator.php

```diff
@@ -0,0 +1,175 @@
+<?php
+/**
+ * This file is part of the PHP Rebilly API package.
+ *
+ * (c) 2015 Rebilly SRL
+ *
+ * For the full copyright and license information, please view the LICENSE.md
+ * file that was distributed with this source code.
+ */
+
+namespace Rebilly;
+
+use ArrayObject;
+use Countable;
+use Iterator;
+use RuntimeException;
+use OutOfBoundsException;
+
+/**
+ * Class Paginator.
+ *
+ * @author Veaceslav Medvedev <veaceslav.medvedev@rebilly.com>
+ * @version 0.1
+ */
+final class Paginator implements Iterator, Countable
+{
+    const DEFAULT_SIZE = 100;
+
+    /** @var Client */
+    private $client;
+
+    /** @var string */
+    private $uri;
+
+    /** @var array */
+    private $params = [];
+
+    /** @var int */
+    private $limit = self::DEFAULT_SIZE;
+
+    /** @var int */
+    private $offset = 0;
+
+    /** @var int */
+    private $totalItems;
+
+    /** @var Rest\Collection */
+    private $currentSegment;
+
+    /**
+     * @param Client $client
+     * @param string $uri
+     * @param array|ArrayObject $params
+     */
+    public function __construct(Client $client, $uri, $params = [])
+    {
+        $this->client = $client;
+        $this->uri = $uri;
+        $this->params = (array) $params + ['limit' => $this->limit, 'offset' => $this->offset];
+
+        $this->load($this->params);
+    }
+
+    /**
+     * @param array $params
+     */
+    private function load(array $params)
+    {
+        /** @var Rest\Collection $segment */
+        $segment = $this->client->get($this->uri, $params);
+
+        if ($segment->getOffset() % $segment->getLimit() !== 0) {
+            throw new RuntimeException('Not relevant segment offset and limit');
+        }
+
+        $this->currentSegment = $segment;
+        $this->totalItems = $segment->getTotalItems();
+        $this->offset = $segment->getOffset();
+        $this->limit = $segment->getLimit();
+    }
+
+    /**
+     * {@inheritdoc}
+     *
+     * @return Rest\Collection
+     */
+    public function current()
+    {
+        if ($this->currentSegment === null) {
+            if (!$this->valid()) {
+                throw new OutOfBoundsException('Cannot load segment, invalid offset');
```

Comment:
in case of this error what message will be shown to the user?

because the user does not know what is a `segment`, `page` would make more sense to the user